src/dune_rules/expander.mli

```diff
@@ -145,3 +145,5 @@ val map_exe : t -> Path.t -> Path.t
 val artifacts : t -> Artifacts.Bin.t
 
 val find_package : t -> Package.Name.t -> any_package option Memo.t
+
+val expand_locks : ?base:Path.t -> t -> Locks.t -> Path.t list Memo.t
```

Comment:
To me, this `?base` argument is a bit confusing. I'd prefer something like:

```
val expand_locks : t -> base:[`Of_expander | `This of Path.t] -> Locks.t -> Path.t list Memo.t
```

Reply:
good idea, done